src/SFA.DAS.AdminService.Web.UnitTests/Controllers/Home/HomeControllerTest.cs

```diff
@@ -49,26 +52,48 @@ public void Then_Index_Returns_ViewModel(bool useDfESignIn)
         public void When_DfESignIn_True_And_User_Authenticated_Then_Index_Returns_Redirect()
         {
             //arrange
-            _configuration.Setup(c => c.UseDfESignIn).Returns(true);
+            _mockWebConfiguration.Setup(c => c.UseDfESignIn).Returns(true);
             var authorizedUser = new ClaimsPrincipal(new ClaimsIdentity(new[]
             {
                 new Claim("name", "somename")
             }, "mock"));
             var httpContext = new Mock<HttpContext>();
             httpContext.Setup(c => c.User).Returns(authorizedUser);
-            var controller = new HomeController(_configuration.Object)
+            _controller = new HomeController(_mockWebConfiguration.Object, _mockConfiguration.Object)
             {
                 ControllerContext = new ControllerContext { HttpContext = httpContext.Object }
             };
 
             //sut
-            var result = controller.Index();
+            var result = _controller.Index();
 
             //assert
             result.Should().NotBeNull();
             var actualResult = (RedirectToActionResult)result;
             actualResult.ControllerName.Should().Be("Dashboard");
             actualResult.ActionName.Should().Be("Index");
         }
+
+        [TestCase("test", "https://test-services.signin.education.gov.uk/approvals/select-organisation?action=request-service", true)]
+        [TestCase("pp", "https://test-services.signin.education.gov.uk/approvals/select-organisation?action=request-service", true)]
+        [TestCase("local", "https://test-services.signin.education.gov.uk/approvals/select-organisation?action=request-service", false)]
+        [TestCase("prd", "https://services.signin.education.gov.uk/approvals/select-organisation?action=request-service", false)]
+        public void When_InvalidRole_Then_ViewIsReturned(string env, string helpLink, bool useDfESignIn)
+        {
+            //arrange
+            _mockWebConfiguration.Setup(x => x.UseDfESignIn).Returns(useDfESignIn);
+            _mockConfiguration.Setup(x => x["ResourceEnvironmentName"]).Returns(env);
+            _controller = new HomeController(_mockWebConfiguration.Object, _mockConfiguration.Object);
+
+            //sut
+            var result = (ViewResult)_controller.InvalidRole();
+
+            //assert
+            Assert.That(result, Is.Not.Null);
+
+            var actualModel = result.Model as Error403ViewModel;
+            Assert.That(actualModel?.HelpPageLink, Is.EqualTo(helpLink));
+            Assert.That(actualModel?.UseDfESignIn, Is.EqualTo(useDfESignIn));
```

Comment:
If one of these asserts fail, the test will abort without running the subsequent asserts. We can make all the asserts run regardless of failure by wrapping this block in an `AssertMultiple`

```suggestion
           Assert.Multiple(() => 
           { 
               Assert.That(result, Is.Not.Null);

               var actualModel = result.Model as Error403ViewModel;
               Assert.That(actualModel?.HelpPageLink, Is.EqualTo(helpLink));
               Assert.That(actualModel?.UseDfESignIn, Is.EqualTo(useDfESignIn));
           });
```